runtime/common/marshaling/index.ts

```diff
@@ -173,12 +174,17 @@ export function marshalValue(val: any): MarshaledValue {
   };
 }
 
-export function marshalValueToString(val: any): string {
+export function marshalValueToStringForSameOrHigherNodeVersion(val: any): string {
   return serialize(marshalValue(val));
 }
 
 export function unmarshalValueFromString(marshaledValue: string): any {
-  return unmarshalValue(deserialize(marshaledValue));
+  if (marshaledValue?.startsWith('/')) {
+    // Looks like a v8-serialized value
+    return unmarshalValue(deserialize(marshaledValue));
```

Comment:
Maybe a try/catch around `deserialize` in case that fails?

Reply:
What error handling were you thinking here?

I don't think we'd want to suppress any errors here for fear of moving incorrect data around. This can't be a legacy marshaled value if it starts with "/" since that can't be the beginning of JSON string.

Reply:
Ah, I suppose that's sufficient then.